src/compiler/scala/tools/nsc/util/StackTracing.scala

```diff
@@ -54,16 +54,26 @@ private[util] trait StackTracing extends Any {
     def print(e: Throwable, r: TraceRelation, share: Array[StackTraceElement], indents: Int): Unit = if (unseen(e)) {
       val trace  = e.getStackTrace
       val frames = if (share.isEmpty) trace else {
-        val spare  = share.reverseIterator
+        val spare   = share.reverseIterator
         val trimmed = trace.reverse dropWhile (spare.hasNext && spare.next == _)
         trimmed.reverse
       }
       val prefix = frames.takeWhile(p)
       val margin = "  " * indents
       lines += s"${margin}${r}${header(e)}"
       prefix.foreach(f => lines += s"${margin}  at $f")
-      if (frames.size < trace.size) lines += s"${margin}  ... ${trace.size - frames.size} more"
-      if (r == Self && prefix.size < frames.size) lines += s"${margin}  ... ${frames.size - prefix.size} elided"
+
+      val traceFramesLenDiff = trace.length - frames.length
+      val framesPrefixLenDiff = frames.length - prefix.length
+      if (traceFramesLenDiff > 0) {
+        if (framesPrefixLenDiff > 0) lines += s"${margin}  ... ${framesPrefixLenDiff} elided and ${traceFramesLenDiff} more"
+        else lines += s"${margin}  ... ${traceFramesLenDiff} more"
+      } else if (framesPrefixLenDiff > 0) {
+        lines += s"${margin}  ... ${framesPrefixLenDiff} elided"
+      }
+
+//      if (frames.length < trace.length) lines += s"${margin}  ... ${trace.length - frames.length} more"
+//      if (r == Self && prefix.length < frames.length) lines += s"${margin}  ... ${frames.length - prefix.length} elided"
       print(e.getCause, CausedBy, trace, indents)
       e.getSuppressed.foreach(print(_, Suppressed, frames, indents + 1))
```

Comment:
Thanks, and sorry you had to crack that open.

Did you prefer one liner or stacked for saying elided and more? Either way, this fix looks good.

I'm not sure why print recurses with trace for cause and frames for suppress. I wonder if there is a reason it says elided only at top level (Self).

Is it true there is no necessary relationship between the stack traces of an exception, its cause, and other exceptions it suppresses? Although suppressions are paradigmatically due to exceptions on close, one could imagine suppressing async errors on a different stack. I wonder if anything in scala.concurrent does that.

I forgot I had touched this in January, before work started up again. I remember I intended to make it readable. Ha.

I'll have a follow-up edit which I'll let you pull if you like, or I'll use it later. There's also a junit test. I was linting junit tests today, so you inspired me to take another whack at this one. For instance, I wanted to compare stack trace and the trimmed representation side-by-side.

Reply:
FYI, `header` is easier to read when aligned. I'll get back to the unit test later.

https://github.com/som-snytt/scala/commit/0009dc9d682f45e4eb87aa16fc5d79c6eef8b7d9

Reply:
> sorry you had to crack that open

I was happy to - I think I learned quite a bit :)

> Did you prefer one liner or stacked for saying elided and more?

I think I prefer the one liner. I only asked because stacked is a much smaller change

> Is it true there is no necessary relationship between the stack traces of an exception, its cause, and other exceptions it suppresses?

pretty much, yeah

> Although suppressions are paradigmatically due to exceptions on close

it doesn't have to be from closing the same resource though. you could have one exception thrown from using one resource, and then another thrown by closing a different resource you had open

> I'll have a follow-up edit which I'll let you pull if you like

I'm happy to make edits if there are issues. I know there are a few things I forgot to clean up after working on it

Reply:
@som-snytt feel free to push changes if you'd like

Reply:
I don't want to be that guy, but thanks for taking the PRETY version of header method.